### ISSUE TITLE
`context.df.isReplaying` not updating between activity calls

### ISSUE THREAD

**Issue by poelstra** [Azure/azure-functions-durable-js#47]
According to e.g. https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-diagnostics#javascript-functions-2x-only-1 and [this video](https://channel9.msdn.com/Shows/On-NET/Debugging-and-Diagnostics-of-Durable-Functions?term=Chris%20Gillum%20-%20Serverless%20workflows%20with%20Durable%20Functions&WT.mc_id=functions-recipes-docs-cephilli#time=04m14s) isReplaying should be a 'live' property that updates before and after each yield to indicate whether that piece of code has been visited before.

This does not seem to be the case.

E.g. with the following orchestrator:
```
const df = require("durable-functions");

module.exports = df.orchestrator(function*(context) {
    if (!context.df.isReplaying) { context.log(`Calling 1`); }
    yield context.df.callActivity("TestActivity", 1);

    if (!context.df.isReplaying) { context.log(`Calling 2`); }
    yield context.df.callActivity("TestActivity", 2);

    if (!context.df.isReplaying) { context.log(`Calling 3`); }
    yield context.df.callActivity("TestActivity", 3);

    if (!context.df.isReplaying) { context.log(`Conditional log`); }
    context.log("Unconditional log");
});
```

I get:

```txt
[8-2-2019 14:02:41] Executing HTTP request: {
[8-2-2019 14:02:41]   "requestId": "262d31ee-ca4b-41f9-9338-1ffd9410dd74",
[8-2-2019 14:02:41]   "method": "POST",
[8-2-2019 14:02:41]   "uri": "/api/orchestrators/TestOrchestrator"
[8-2-2019 14:02:41] }
[8-2-2019 14:02:41] Executing 'Functions.HttpStart' (Reason='This function was programmatically called via the host APIs.', Id=fc8069de-128e-408b-8095-b9ea313bb235)
[8-2-2019 14:02:41] Executing HTTP request: {
[8-2-2019 14:02:41]   "requestId": "290b3408-15b0-4d95-af30-6bb734f467a2",
[8-2-2019 14:02:41]   "method": "POST",
[8-2-2019 14:02:41]   "uri": "/runtime/webhooks/durabletask/orchestrators/TestOrchestrator"
[8-2-2019 14:02:41] }
[8-2-2019 14:02:42] 9287fec086514b6d996df2b48dfd368c: Function 'TestOrchestrator (Orchestrator)' scheduled. Reason: NewInstance. IsReplay: False. State: Scheduled. HubName: DurableFunctionsHub. AppName: . SlotName: . ExtensionVersion: 1.7.0. SequenceNumber: 2.
[8-2-2019 14:02:42] Executed HTTP request: {
[8-2-2019 14:02:42]   "requestId": "290b3408-15b0-4d95-af30-6bb734f467a2",
[8-2-2019 14:02:42]   "method": "POST",
[8-2-2019 14:02:42]   "uri": "/runtime/webhooks/durabletask/orchestrators/TestOrchestrator",
[8-2-2019 14:02:42]   "identities": [
[8-2-2019 14:02:42]     {
[8-2-2019 14:02:42]       "type": "WebJobsAuthLevel",
[8-2-2019 14:02:42]       "level": "Admin"
[8-2-2019 14:02:42]     }
[8-2-2019 14:02:42] Started orchestration with ID = '9287fec086514b6d996df2b48dfd368c'.
[8-2-2019 14:02:42]   ],
[8-2-2019 14:02:42]   "status": 202,
[8-2-2019 14:02:42]   "duration": 521
[8-2-2019 14:02:42] }
[8-2-2019 14:02:42] Executed 'Functions.HttpStart' (Succeeded, Id=fc8069de-128e-408b-8095-b9ea313bb235)
[8-2-2019 14:02:42] Executed HTTP request: {
[8-2-2019 14:02:42]   "requestId": "262d31ee-ca4b-41f9-9338-1ffd9410dd74",
[8-2-2019 14:02:42]   "method": "POST",
[8-2-2019 14:02:42]   "uri": "/api/orchestrators/TestOrchestrator",
[8-2-2019 14:02:42]   "identities": [
[8-2-2019 14:02:42]     {
[8-2-2019 14:02:42]       "type": "WebJobsAuthLevel",
[8-2-2019 14:02:42]       "level": "Admin"
[8-2-2019 14:02:42]     }
[8-2-2019 14:02:42]   ],
[8-2-2019 14:02:42]   "status": 202,
[8-2-2019 14:02:42]   "duration": 868
[8-2-2019 14:02:42] }
[8-2-2019 14:02:42] Host lock lease acquired by instance ID '000000000000000000000000A2BDC8CD'.
[8-2-2019 14:02:58] Executing 'Functions.TestOrchestrator' (Reason='', Id=b2efbbe3-3160-4f06-b1c5-334ff8b9822b)
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestOrchestrator (Orchestrator)' started. IsReplay: False. Input: (16 bytes). State: Started. HubName: DurableFunctionsHub. AppName: . SlotName: . ExtensionVersion: 1.7.0. SequenceNumber: 3.
[8-2-2019 14:02:58] Calling 1
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestActivity (Activity)' scheduled. Reason: TestOrchestrator. IsReplay: False. State: Scheduled. HubName: DurableFunctionsHub. AppName: . SlotName: . ExtensionVersion: 1.7.0. SequenceNumber: 4.
[8-2-2019 14:02:58] Executed 'Functions.TestOrchestrator' (Succeeded, Id=b2efbbe3-3160-4f06-b1c5-334ff8b9822b)
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestActivity (Activity)' started. IsReplay: False. Input: (12 bytes). State: Started. HubName: DurableFunctionsHub. AppName: . SlotName: . ExtensionVersion: 1.7.0. SequenceNumber: 5.
[8-2-2019 14:02:58] Executing 'Functions.TestActivity' (Reason='', Id=984dba6b-b196-4dc2-801a-901cab3da3bc)
[8-2-2019 14:02:58] Executed 'Functions.TestActivity' (Succeeded, Id=984dba6b-b196-4dc2-801a-901cab3da3bc)
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestActivity (Activity)' completed. ContinuedAsNew: False. IsReplay: False. Output: (4 bytes). State: Completed. HubName: DurableFunctionsHub. AppName: . SlotName: .
ExtensionVersion: 1.7.0. SequenceNumber: 6.
[8-2-2019 14:02:58] Executing 'Functions.TestOrchestrator' (Reason='', Id=87ba9990-0b31-4566-82e4-cc06927d1d69)
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestActivity (Activity)' scheduled. Reason: TestOrchestrator. IsReplay: False. State: Scheduled. HubName: DurableFunctionsHub. AppName: . SlotName: . ExtensionVersion: 1.7.0. SequenceNumber: 7.
[8-2-2019 14:02:58] Executed 'Functions.TestOrchestrator' (Succeeded, Id=87ba9990-0b31-4566-82e4-cc06927d1d69)
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestActivity (Activity)' started. IsReplay: False. Input: (12 bytes). State: Started. HubName: DurableFunctionsHub. AppName: . SlotName: . ExtensionVersion: 1.7.0. SequenceNumber: 8.
[8-2-2019 14:02:58] Executing 'Functions.TestActivity' (Reason='', Id=5787431a-0daa-44eb-8fdd-1e326e6ee3d6)
[8-2-2019 14:02:58] Executed 'Functions.TestActivity' (Succeeded, Id=5787431a-0daa-44eb-8fdd-1e326e6ee3d6)
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestActivity (Activity)' completed. ContinuedAsNew: False. IsReplay: False. Output: (4 bytes). State: Completed. HubName: DurableFunctionsHub. AppName: . SlotName: .
ExtensionVersion: 1.7.0. SequenceNumber: 9.
[8-2-2019 14:02:58] Executing 'Functions.TestOrchestrator' (Reason='', Id=57e14dfa-ad64-4059-868d-c189900cc2ae)
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestActivity (Activity)' scheduled. Reason: TestOrchestrator. IsReplay: False. State: Scheduled. HubName: DurableFunctionsHub. AppName: . SlotName: . ExtensionVersion: 1.7.0. SequenceNumber: 10.
[8-2-2019 14:02:58] Executed 'Functions.TestOrchestrator' (Succeeded, Id=57e14dfa-ad64-4059-868d-c189900cc2ae)
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestActivity (Activity)' started. IsReplay: False. Input: (12 bytes). State: Started. HubName: DurableFunctionsHub. AppName: . SlotName: . ExtensionVersion: 1.7.0. SequenceNumber: 11.
[8-2-2019 14:02:58] Executing 'Functions.TestActivity' (Reason='', Id=d060ffad-5161-4552-900b-d95cfce72005)
[8-2-2019 14:02:58] Executed 'Functions.TestActivity' (Succeeded, Id=d060ffad-5161-4552-900b-d95cfce72005)
[8-2-2019 14:02:58] 9287fec086514b6d996df2b48dfd368c: Function 'TestActivity (Activity)' completed. ContinuedAsNew: False. IsReplay: False. Output: (4 bytes). State: Completed. HubName: DurableFunctionsHub. AppName: . SlotName: .
ExtensionVersion: 1.7.0. SequenceNumber: 12.
[8-2-2019 14:02:58] Executing 'Functions.TestOrchestrator' (Reason='', Id=0d8e5c1d-1308-4d61-8a08-51faeef50404)
[8-2-2019 14:02:58] Unconditional log
[8-2-2019 14:02:58] Executed 'Functions.TestOrchestrator' (Succeeded, Id=0d8e5c1d-1308-4d61-8a08-51faeef50404)
```

Note that there's only "Calling 1" and "Unconditional log", I would have expected each log line to be present exactly once.

**Comment by sdoucetOctopus** [comment]
Hi,

Any update on this issue ?

Thanks

**Comment by kashimiz** [comment]
Hi @sdoucetOctopus, my apologies for the late response. we were unable to include a fix for this issue in the latest release (1.2.2), but I've been able to repro it and will be adding a fix in the next release.

**Comment by davidmrdavid** [comment]
Closing as this has been released
